### DISCUSSION TITLE
Open WebUI and NVIDIA NIM Intergration

### DISCUSSION THREAD

**Discussion by yuchenxie4645** [open-webui/open-webui#2859] [Show and tell]
I manage to figure out how I can intergrate NVIDIA NIM API onto Open WebUI, all you have to do is replace the OpenAI API website with NVIDIA's API website and generate your own key. It is currently free with upto 1000 API call request and up to 5000 for corporate email.

**Comment by justinh-rahb** [comment]
Woah... that's a lot of new models to access now!

![Screenshot 2024-06-05 at 1 49 26 PM](https://github.com/open-webui/open-webui/assets/52832301/618e1682-0a78-4cc4-b469-1ba87db1d3d8)

**Reply by yuchenxie4645** [reply]
I took the time, and labeled all of them :D
![image](https://github.com/open-webui/open-webui/assets/169230870/e24f00c5-60f2-4df1-9ccc-8440dda11d59)

**Comment by justinh-rahb** [comment]
Hmmmm... my enthusiasm may have been premature. I can only get a small handful of models to output a ***useful*** amount of tokens. It seems to me like the basic API key is only intended to test the connectivity of the API, not for actual usage (even in a demonstrative capacity).

**Reply by yuchenxie4645** [reply]
Becuase some models listed are not even available on NVIDIAs NIMs page. And some of them are dysfunctional (only outputs like 5 tokens) but some models, like mixtral, and Llama 70b is entirely functional.

**Reply by justinh-rahb** [reply]
Yes, I found the list of functioning models to be quite small. Codegemma, Gemma 1.1 7B, Llama3, Mixtral 8x22B, and the Phi3 models basically.

**Reply by yuchenxie4645** [reply]
I mean, Mixtral 8x22B is really 176B parameter model, and Llama3 70B is among one of the most accurate ones in NIMs, so I wouldn't complain.

**Reply by justinh-rahb** [reply]
> I mean, Mixtral 8x22B is really 176B parameter model, and Llama3 70B is among one of the most accurate ones in NIMs, so I wouldn't complain.

This is true, but I already have paid API access for these. Would have been nice to try some of the ones I don't have already, but none of them worked. It's alright, I'm sure it's a fine service as long as you're paying for it. Workplace policy would mean I can't use these free APIs for anything serious anyway.
